src/passes/OptimizeInstructions.cpp

```diff
@@ -974,25 +955,50 @@ struct OptimizeInstructions
   Expression* optimizeBoolean(Expression* boolean) {
     // TODO use a general getFallthroughs
     if (auto* unary = boolean->dynCast<Unary>()) {
-      if (unary && unary->op == EqZInt32) {
-        auto* unary2 = unary->value->dynCast<Unary>();
-        if (unary2 && unary2->op == EqZInt32) {
-          // double eqz
-          return unary2->value;
+      if (unary) {
+        if (unary->op == EqZInt32 || unary->op == EqZInt64) {
+          if (unary->op == EqZInt32) {
+            auto* unary2 = unary->value->dynCast<Unary>();
+            if (unary2 && unary2->op == EqZInt32) {
+              // double eqz
+              return unary2->value;
+            }
+          }
+          if (auto* c = unary->value->dynCast<Const>()) {
```

Comment:
I think this is unnecessary. An eqz of a constant will be optimized by the Precompute pass anyhow.

Reply:
Unfortunately it can't handle many cases like:
```wat
(i64.eqz
          (select
           (i64.const 58)
           (i64.const -982757)
           (i64.eqz
            (i64.const 0)
           )
          )
         )
```
in `optimize-instructions_optimize-level=2_ignore-implicit-traps.txt`
and other places as well. So I'm not sure Precompute handle this as expected

Reply:
Alright I was under impression `optimize-instructions_optimize-level=2...` use precompute-propagate but it use just `precompute` pass so with first one I guess all should already handled so removed this const evaluation part